cocoa/core/controller.py

```diff
@@ -53,14 +57,14 @@ def simulate(self, max_turns=None):
                 if self.debug:
                     print 'agent=%s, session=%s, event=%s' % (agent, type(session).__name__, event.to_dict())
                 else:
-                    event_dict = event.to_dict()
-                    action = event_dict['action']
-                    data = event_dict['data']
+                    action = event.action
+                    data = event.data
                     event_output = data if action == 'message' else "Action: {0}, Data: {1}".format(action, data)
                     print 'agent=%s, event=%s' % (agent, event_output)
                 num_turns += 1
                 if self.game_over() or (max_turns and num_turns >= max_turns):
                     game_over = True
+                    self.postgame_check(num_turns, max_turns)
```

Comment:
Should postgame_check be game_over instead?

Reply:
I moved `postgame_check` into the `game_over` method, so that it occurs whenever we know the game is about to end.  Does that fit better with what you were thinking?

Just to give some context, I originally had it as part of `simulate` because I thought it might be useful for other systems to also have a "cleanup" section distinct from `game_over`, which would be restricted to only identifying when the chat has ended.